### PR TITLE
87 read interval type

### PR DESCRIPTION
**Pull request by azurecoder**
### Fixes

Issue #87

### Description

Supports the reading and writing of interval types. Not yet implemented in Spark so followed Parquet spec months (4 bytes LE) / days (4 bytes LE) / millis (4 bytes LE) with annotated INTERVAL on FIXED_LEN_BYTE_ARRAY

### Todos
- [ ] unit/integration tests
- [ ] documentation

### CODE REVIEW

**Comment by azurecoder** [comment]
Done a few refactorings around SchemaElement to make checks with derived types easier - less spaghetti code now

File: src/Parquet/File/TypeFactory.cs [hunk 1]
```diff
@@ -128,12 +129,15 @@ public static Type ToSystemType(Thrift.SchemaElement schema)
             case Thrift.Type.DOUBLE:
                return typeof(double);
             case Thrift.Type.INT96:
```
Comment (xhochy) [inline src/Parquet/File/TypeFactory.cs]: Note that INT96 is never annotated with a ConvertedType. It's only ever use was for nanosecond precision timestamps that were made up of the julian day + the nanoseconds in that day.

Comment (azurecoder) [inline src/Parquet/File/TypeFactory.cs]: correct there is no annotation, we don't use a convertedtype and follow the Julian calendar in days, 
`byte[] v96 = new byte[4];
               byte[] nanos = new byte[8];
               Array.Copy(data, i + 8, v96, 0, 4);
               Array.Copy(data, i, nanos, 0, 8);
               DateTime bi = (BitConverter.ToInt32(v96, 0) - 1).JulianToDateTime();
               long nanosToInt64 = BitConverter.ToInt64(nanos, 0);
               double millis = (double) nanosToInt64 / 1000000D;
               bi = bi.AddMilliseconds(millis);
               destination.Add(new DateTimeOffset(bi));`

Comment (azurecoder) [inline src/Parquet/File/TypeFactory.cs]: Unfortunately per the above we can't support the nano-second precision anyway in the type we're surfacing in C#.

**Pull request merged**